src/root.zig

```diff
@@ -100,6 +107,12 @@ pub fn Algorithm(
 
         impl: Impl,
 
+        const TypeOfCaracter = switch (Impl) {
+            AsciiOptions => u8,
+            UnicodeOptions => u21,
+            else => unreachable,
+        };
```

Comment:
The `Algorithm` generic struct already has an `ElType` argument, so we can use that instead of having to maintain a switch. In theory I still want downstream users to be able to modify the behaviour of the fuzzy finder without having to modify the source.

Reply:
For me, there are two types of solutions:
1. The user should give a standard string (`[]const u8`) to the fuzzy finder, choose the right algo (Ascii or Unicode) depending on the “type” of the text. In this case, `ElType` will always be `u8`, and `TypeOfCaracter` depends on the algo. The algo will make by himself all the conversion.
2. Let the user make the conversions, and the algo will take the type of the strings (`ElType`). The problem is that the user will need a lib to make the conversion, for being able to use fuzzig. And fuzzig will also need a lib (surely the same) to process correctly the data.

The solution for me is the 1, to replace `ElType` by `u8` where it is needed, and use it instead to provide the type that is waited by the `UnicodeOption` (or Ascii).
In this case, `TypeOfCaracter` is no longer needed (so we can delete the switch).

Reply:
I have done it in https://github.com/alberic89/fuzzig/commit/073d10a8f4b2ae945392e1cf059b4af85b3066a1